Initial changes required to run the Shoreline plug-ins on viz003

Jeff-Hadley opened the pull request:
Changed lib IGL paths, added CMake compiler options and fields. Made changes to find the correct BoostConfig.cmake file for the createHarmonicSkeleton plugin.  Changes were necessary to run the plug-ins on CU Boulder's PHASTA group viz-nodes.

Commit:
Initial changes required to run the Shoreline plug-ins on viz003

	modified:   ParaViewPlugins/createHarmonicSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/vtkcreateSkeletonFilter.cxx
	modified:   ParaViewPlugins/interactiveSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/vtkInteractiveSkeletonFilter.cxx
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/vtkMySkeletonRepresentation.cxx
	modified:   ParaViewPlugins/scaleDirection/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/scaleNormal/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/translateFeature/Plugin/Filters/CMakeLists.txt

	modified:   ParaViewPlugins/createHarmonicSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/vtkcreateSkeletonFilter.cxx
	modified:   ParaViewPlugins/interactiveSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/scaleDirection/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/scaleNormal/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/translateFeature/Plugin/Filters/CMakeLists.txt
	deleted:    ClientServerManagers/build/CMakeCache.txt
	deleted:    ClientServerManagers/build/CMakeFiles/3.16.0/CMakeCCompiler.cmake
	deleted:    ClientServerManagers/build/CMakeFiles/3.16.0/CMakeCXXCompiler.cmake
	deleted:    ClientServerManagers/build/CMakeFiles/3.16.0/CMakeDetermineCompilerABI_C.bin
	deleted:    ClientServerManagers/build/CMakeFiles/3.16.0/CMakeDetermineCompilerABI_CXX.bin
	deleted:    ClientServerManagers/build/CMakeFiles/3.16.0/CMakeSystem.cmake
	deleted:    ClientServerManagers/build/CMakeFiles/3.16.0/CompilerIdC/CMakeCCompilerId.c
	deleted:    ClientServerManagers/build/CMakeFiles/3.16.0/CompilerIdC/a.out
	deleted:    ClientServerManagers/build/CMakeFiles/3.16.0/CompilerIdCXX/CMakeCXXCompilerId.cpp
	deleted:    ClientServerManagers/build/CMakeFiles/3.16.0/CompilerIdCXX/a.out
	deleted:    ClientServerManagers/build/CMakeFiles/CMakeOutput.log
	deleted:    ClientServerManagers/build/CMakeFiles/cmake.check_cache
	deleted:    cache/skeleton.vtp
	deleted:    cache/skeletonMod.vtp

	modified:   ParaViewPlugins/createHarmonicSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/vtkcreateSkeletonFilter.cxx
	modified:   ParaViewPlugins/interactiveSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/vtkInteractiveSkeletonFilter.cxx
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/vtkMySkeletonRepresentation.cxx
	modified:   ParaViewPlugins/scaleDirection/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/scaleNormal/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/translateFeature/Plugin/Filters/CMakeLists.txt

	modified:   ParaViewPlugins/createHarmonicSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/vtkcreateSkeletonFilter.cxx
	modified:   ParaViewPlugins/interactiveSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/vtkInteractiveSkeletonFilter.cxx
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/vtkMySkeletonRepresentation.cxx
	modified:   ParaViewPlugins/scaleDirection/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/scaleNormal/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/translateFeature/Plugin/Filters/CMakeLists.txt
	modified:   ClientServerManagers/shClient.cpp
	modified:   ClientServerManagers/src/MeshCourier.cpp
	modified:   ClientServerManagers/src/MeshCourierClient.cpp
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/CMakeLists.txt

	modified:   ParaViewPlugins/createHarmonicSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/CMakeLists.txt
	modified:   ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/vtkcreateSkeletonFilter.cxx
	modified:   ParaViewPlugins/interactiveSkeleton/CMakeLists.txt
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/vtkInteractiveSkeletonFilter.cxx
	modified:   ParaViewPlugins/interactiveSkeleton/Plugin/Filters/vtkMySkeletonRepresentation.cxx
	modified:   ParaViewPlugins/scaleDirection/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/scaleNormal/Plugin/Filters/CMakeLists.txt
	modified:   ParaViewPlugins/translateFeature/Plugin/Filters/CMakeLists.txt

	modified:   ../../CMakeLists.txt
	modified:   CMakeLists.txt
	modified:   vtkcreateSkeletonFilter.cxx
	modified:   ../../../interactiveSkeleton/CMakeLists.txt
	modified:   ../../../interactiveSkeleton/Plugin/Filters/CMakeLists.txt
	modified:   ../../../interactiveSkeleton/Plugin/Filters/vtkInteractiveSkeletonFilter.cxx
	modified:   ../../../interactiveSkeleton/Plugin/Filters/vtkMySkeletonRepresentation.cxx
	modified:   ../../../scaleDirection/Plugin/Filters/CMakeLists.txt
	modified:   ../../../scaleNormal/Plugin/Filters/CMakeLists.txt
	modified:   ../../../translateFeature/Plugin/Filters/CMakeLists.txt
	modified:   ../../../../ClientServerManagers/shClient.cpp
	modified:   ../../../../ClientServerManagers/src/MeshCourier.cpp
	modified:   ../../../../ClientServerManagers/src/MeshCourierClient.cpp
	modified:   CMakeLists.txt
	modified:   ../../../interactiveSkeleton/Plugin/Filters/CMakeLists.txt

nelson15 commented on ParaViewPlugins/createHarmonicSkeleton/CMakeFiles/cmake.check_cache:
```diff
@@ -0,0 +1 @@
+# This file is generated by cmake for dependency checking of the CMakeCache.txt file
```
Can you remove this file?

Jeff-Hadley replied:
Removed.

nelson15 commented on ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/CMakeLists.txt:
```diff
@@ -9,15 +9,14 @@ vtk_module_add_module(createSkeletonFilters
+find_package(Boost REQUIRED) 
 # find_package (Eigen3 3.3 REQUIRED NO_MODULE)
-add_definitions("-DCGAL_EIGEN3_ENABLED")#-DCGAL_HEADER_ONLY
+# add_definitions("-DCGAL_EIGEN3_ENABLED")#-DCGAL_HEADER_ONLY ***Commented this out***
```
Just delete this line

nelson15 replied:
You can delete this comment

Jeff-Hadley replied:
Should all be cleaned up now.

nelson15 commented on ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/CMakeLists.txt:
```diff
@@ -9,15 +9,14 @@ vtk_module_add_module(createSkeletonFilters
 
 vtk_module_include( createSkeletonFilters
-  PRIVATE "/Users/coreylee/Git/libigl/include"
+  PRIVATE "/users/jeffhadley/Builds/src/libigl/include"
```
We gotta figure out why libigl won't link...

nelson15 commented on ParaViewPlugins/createHarmonicSkeleton/Plugin/createSkeletonFilters/CMakeLists.txt:
```diff
@@ -9,15 +9,14 @@ vtk_module_add_module(createSkeletonFilters
-)
-
+vtk_module_link( createSkeletonFilters PRIVATE CGAL::CGAL Eigen3::Eigen igl::core Boost::boost)
+	#PRIVATE CGAL::CGAL Eigen3::Eigen igl::core ${Boost_LIBRARIES})
```
delete this commented out line

Jeff-Hadley replied:
Removed.

nelson15 commented on ParaViewPlugins/interactiveSkeleton/Plugin/Filters/CMakeLists.txt:
```diff
@@ -19,6 +19,6 @@ find_package (Eigen3 3.3 REQUIRED NO_MODULE)
-vtk_module_include(interactiveSkeletonFilters PUBLIC ${EMBREE_INCLUDES} "/Users/coreylee/Git/libigl_Test/libigl/external/embree/include" "/Users/coreylee/Git/libigl_Test/build/include")
-
+# vtk_module_include(interactiveSkeletonFilters PUBLIC ${EMBREE_INCLUDES} "/Users/coreylee/Git/libigl_Test/libigl/external/embree/include" "/Users/coreylee/Git/libigl_Test/build/include")
+vtk_module_include(interactiveSkeletonFilters PUBLIC ${EMBREE_INCLUDES} "/users/jeffhadley/Builds/src/libigl/include/igl/embree" "/users/jeffhadley/Builds/src/libigl/include") 
```
can you test removing this line? I'm not entirely sure its needed

nelson15 replied:
for background, Embree is a raytracing library and I was experimenting with using a raytracing algorithm for automatic skeleton skinning weights, but I never could get it working.

Jeff-Hadley replied:
It still works without it in there. Just pushed the changes.

nelson15 replied:
You can delete that line. And remove the Embree stuff

Jeff-Hadley replied:
Should all be cleaned up now.

nelson15 reviewed (commented):
Can you remove all the files with the *.un~ suffix? Not sure where those came from (maybe an editor lock file?) and we gotta figure out how to link libigl properly. that is bananas (not your fault. its just frustrating we don't have a good solution).

Commit:
Additional changes to ClientServerManager files required to run on viz003, as well as clean up of commented lines.
	modified:   ../../../../ClientServerManagers/shClient.cpp
	modified:   ../../../../ClientServerManagers/src/MeshCourier.cpp
	modified:   ../../../../ClientServerManagers/src/MeshCourierClient.cpp
	modified:   CMakeLists.txt
	modified:   ../../../interactiveSkeleton/Plugin/Filters/CMakeLists.txt

nelson15 commented:
thanks for the PR!
